src/raglite/_split_chunks.py

```diff
@@ -68,7 +68,7 @@ def split_chunks(  # noqa: C901, PLR0915
     data = []
     for i in range(len(sentences) - 1):
         r = sentence_length_cumsum[i - 1] if i > 0 else 0
-        idx = np.searchsorted(sentence_length_cumsum - r, max_size)
+        idx = np.searchsorted(sentence_length_cumsum - r, max_size + 1)
```

Comment:
From the docs, I think it might be better to instead use:
```suggestion
        idx = np.searchsorted(sentence_length_cumsum - r, max_size, side="right")
```

Because we do actually want to find the 'right-most' index as the docs say (because that will yield fewer constraints), and it also solves the off-by-one issue.

<img width="759" alt="image" src="https://github.com/user-attachments/assets/b8e05f0e-48fa-4c19-bc1d-e3bece29c0d0" />